services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/DefaultS3CrtAsyncClient.java

```diff
@@ -188,7 +193,29 @@ private static void validateOverrideConfiguration(SdkRequest request) {
                 if (overrideConfiguration.credentialsProvider().isPresent()) {
                     throw new UnsupportedOperationException("Request-level credentials override is not supported");
                 }
+
+                if (overrideConfiguration.metricPublishers() != null) {
+                    throw new UnsupportedOperationException("Request-level Metric Publishers override is not supported");
+                }
+
+                if (overrideConfiguration.apiCallAttemptTimeout().isPresent()) {
+                    throw new UnsupportedOperationException("Request-level apiCallAttemptTimeout override is not supported");
+                }
+
+                if (overrideConfiguration.executionAttributes() != null) {
+                    throw new UnsupportedOperationException("Request-level executionAttributes override is not supported");
+                }
             }
         }
     }
+
+    private static void validateCrtInClassPath() {
+        try {
+            ClassLoaderHelper.loadClass(CRT_CLIENT_CLASSPATH, false, (Class) null);
+        } catch (ClassNotFoundException e) {
+
+            throw new IllegalStateException("Could not load classes from crt library."
+                                            + " 'crt'  module must be on the class path. ", e);
```

Comment:
Maybe something like "Could not load classes from AWS Common Runtime (CRT) library. Make sure you have software.amazon.awssdk.crt:crt on the class path"?